src/array/cuda/labor_sampling.cu

```diff
@@ -247,21 +294,19 @@ __global__ void _CSRRowWiseLayerSampleDegreeKernel(
   constexpr FloatType ONE = 1;
 
   while (out_row < last_row) {
-    const auto row = rows[out_row];
-
-    const auto in_row_start = indptr[row];
+    const auto in_row_start = indptr[out_row];
     const auto out_row_start = subindptr[out_row];
 
-    const IdType degree = indptr[row + 1] - in_row_start;
+    const IdType degree = subindptr[out_row + 1] - out_row_start;
 
     if (degree > 0) {
       // stands for k in in arXiv:2210.13339, i.e. fanout
       const auto k = min(num_picks, degree);
       // slightly better than NS
-      const FloatType d_ = ds ? ds[row] : degree;
+      const FloatType d_ = ds ? ds[out_row] : degree;
```

Comment:
ds and d2s has shape num_rows so this was bugged.

Reply:
Thanks for catching this, can you help us why this was not caught by the existing unit test?

Reply:
In my initial PR, there were no unit tests, and I wasn't reminded about them. I tested the implementation with example training scripts myself. But it turns out there are some scenarios with weights and different edge types with minor bugs. So, I wanted to fix them in this PR and do a sanity check so that at least the different scenarios work properly. This doesn't test for full correctness as that would require statistical testing on the degrees of the sampled subgraphs.

Reply:
I could potentially contribute the statistical testing part in another PR if you think that is okay.

Reply:
statistical testing is pretty hard, might worth to do it offline, and link the result in PR, but building a CI for it is not easy.

Reply:
It is actually not that hard. If there are `n` seed nodes in `S`, then the expected number of edges sampled will be `\sum_{s \in S} \min(k, d_s) ~ nk`. Then, the actual number of sampled edges will deviate by around the square root of the expected number. So, we might not need something complicated to test `importance_sampling=0`. If there are a lot of seed nodes, that is already enough to do the statistical test.